File: js/creditCheck.js
exports.creditCheck = function (str) {
  let arr = [];
  str = str.toString().split("").reverse();
  for (let i in str) {
    if (i % 2 !== 0) {
      arr.push(+str[i] * 2);
    } else {
      arr.push(+str[i]);
    }
  }
  for (let j in arr) {
    if (arr[j] > 9) {
      let bigger = arr[j].toString().split("");
      arr[j] = +bigger[0] + +bigger[1];
    }
  }
  let res = arr.reduce((curr, pre) => curr + pre, 0);
  if (res % 10 === 0) {
    return "The number is valid!";
  } else return "The number is invalid!";
};

// console.log(this.creditCheck("5541808923795240"));

File: python/credit_check.py
import functools

def credit_check(nums):
    arr = []
    nums = list(str(nums)[::-1])
    for i in range(len(nums)):
        if (i % 2 != 0) :
            arr.append(int(nums[i]) * 2)
        else:
            arr.append(int(nums[i]))
    for j in range(len(arr)) :   
        if arr[j] > 9:
            bigger = list(str(arr[j]))
            arr[j] = int(bigger[0]) + int(bigger[1])
            
    res = functools.reduce(lambda a, b : a + b, arr)   
    if res % 10 == 0 :
        return "The number is valid!"
    else : return "The number is invalid!"

    
  

# Your Luhn Algorithm Here
# Expected Output:
# If it is valid, print "The number is valid!"
# If it is invalid, print "The number is invalid!"
print(credit_check(5541808923795240))
